startobs is getting set to 1 in two places and ignoring the hash

dstopp opened the pull request:
-commented out on the init
-initialized lastSelector before it is used on opus init

dstopp committed:
startobs is getting set to 1 in two places and ignoring the hash
-commented out on the init
-initialized lastSelector before it is used on opus init

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into #219_startobs_ignored

dstopp committed:
-handle the corner case where startobs is changed in the hash to an invalid number

rfrenchseti commented on opus/application/static_media/js/browse.js:
```diff
@@ -1177,8 +1177,17 @@ var o_browse = {
+                    opus.prefs[o_browse.getStartObsLabel()] = 1;
+                    o_hash.updateHash();
+                    $("#galleryViewContents").addClass("op-disabled");
+                    $(`${tab} ${contentsView}`).infiniteScroll("loadNextPage");
```
I think you need to add a .loader hide() here. If you drag the slider to the very end, the page loads and then the spinner doesn't go away.

dstopp committed:
Merge branch 'new_ui' of https://github.com/SETI/pds-opus into #219_startobs_ignored

rfrenchseti committed:
Fix dataimages.json to return cart count when view=cart and rename result_count to total_obs_count in JSON return.

dstopp committed:
-rename resultCount and cartCount to totalObsCount
-fix infinite scroll issue at end so that it doesn't try to retrieve more and leave the loader hanging

dstopp committed:
Merge branch 'dataimages_result_count_fix' of https://github.com/rfrenchseti/pds-opus into #219_startobs_ignored

# Conflicts:
#	opus/application/static_media/js/browse.js

rfrenchseti commented on opus/application/static_media/js/cart.js:
```diff
@@ -16,7 +16,7 @@ var o_cart = {
     lastRequestNo: 0,
     downloadInProcess: false,
-    cartCount: 0,
+    totalObsCount : undefined,
```
I tried to group the variables that are common to o_browse and o_cart at the top of the var declaration. I think it would be nice to keep them together. Also could add a comment saying these are in common, and another comment marking the start of the ones that are not in common.

rfrenchseti commented on opus/application/static_media/js/opus.js:
```diff
@@ -713,6 +712,8 @@ var opus = {
         o_widgets.updateWidgetCookies();
 
+        // set these to the current hash on opus init
+        [opus.lastSelections, opus.lastExtras] = o_hash.getSelectionsExtrasFromHash();
```
Could you add the assignment we discussed making force_load = true so it's obvious what's going on?

dstopp replied:
added w/comment

rfrenchseti commented on opus/application/static_media/js/browse.js:
```diff
@@ -1486,8 +1498,12 @@ var o_browse = {
+                        let path = o_browse.getDataURL(view, obsNum, customizedLimitNum);
+                        return path;
+                    }
+                    // returning no value indicates end of infinite scroll...
```
When you hit the end of the observations, the console shows this error:
`infinite-scroll.pkgd.js:1265 GET http://127.0.0.1:8000/undefined 404 (Not Found)`
From that point on infinite scroll fails to work. I think Dave discovered that the returning of "undefined" doesn't work as documented. That's why we were still retrieving extra data even at the end, causing multiple API calls. He hadn't figured out a way around it. The fix you put in breaks scrolling entirely. It would be better to just have the multiple API calls for now.

dstopp replied:
hmm.  I can see that this funciton from infinite-scroll:
function request( url, responseType, onLoad, onError, onLast ) {
  var req = new XMLHttpRequest();
  req.open( 'GET', url, true );
appears to not bail when url is empty, thus being an issue.
I have posted a note to GIT regarding this; i can remove and note but this will also cause side effects.

dstopp committed:
-grouped globals together so cart/browse are consist in define
-added forceLoad
-commented out infinite scroll code that hits a bug in that api
